hive-bigquery-connector-common/src/main/java/com/google/cloud/hive/bigquery/connector/BigQueryStorageHandlerBase.java

```diff
@@ -166,6 +167,13 @@ public void configureOutputJobProperties(TableDesc tableDesc, Map<String, String
     // A workaround for mr mode, as MapRedTask.execute resets mapred.output.committer.class
     conf.set(HiveBigQueryConfig.THIS_IS_AN_OUTPUT_JOB, "true");
 
+    if (HiveUtils.isSparkJob(conf)) {
+      // Spark uses the new "mapreduce" Hadoop API for the job output format
+      conf.set("mapreduce.job.outputformat.class", MapReduceOutputFormat.class.getName());
+      tableDesc.setOutputFileFormatClass(BigQueryOutputFormat.class);
```

Comment:
Why are both MapReduceOutputFormat and BigQueryOutputFormat used?

Reply:
Actually, looks like we don't need to explicitly set BigQueryOutputFormat here because it's already set elsewhere.

But, to answer you question, Spark SQL uses MapReduceOutputFormat for the output committer, and BigQueryOutputFormat for the output writer. I don't really know exactly why they chose to split it that way...